### PR TITLE
HermesClientBasicRetryCondition should handle SocketTimeoutException …

### PR DESCRIPTION
**Pull request by macieg**
Implemented …#1036

### CODE REVIEW

File: hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBasicRetryCondition.java [hunk 1]
```diff
@@ -1,13 +1,18 @@
...
 import static java.net.HttpURLConnection.HTTP_CLIENT_TIMEOUT;
 
 public class HermesClientBasicRetryCondition implements Predicate<HermesResponse> {
     @Override
     public boolean test(HermesResponse response) {
-        return response == null || (isClientTimeoutOrServerError(response) || isFailedExceptionally(response));
+        return response == null
+            || isClientTimeoutOrServerError(response)
+            || isFailedExceptionally(response)
+            || isSocketTimeoutException(response.getFailureCause());
```
Comment (cristaloleg) [inline hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBasicRetryCondition.java]: To keep the code more common, maybe we can change `isSocketTimeoutException` param to the `HermesResponse` ? WDYT?

Comment (macieg) [reply hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBasicRetryCondition.java]: I agree

**Comment by cristaloleg** [comment]
Care to add tests? 😉

**Comment by macieg** [comment]
This change is pointless, _isFailedExceptionally_ check includes SocketTimeoutException.

**Pull request closed without merging**